test/auto-languageclient.test.ts

```diff
@@ -12,20 +19,91 @@ function mockEditor(uri: string, scopeName: string): any {
   }
 }
 
+function setupClient() {
+  atom.workspace.getTextEditors().forEach((editor) => editor.destroy())
+  atom.project.getPaths().forEach((project) => atom.project.removePath(project))
+  const client = new FakeAutoLanguageClient()
+  client.activate()
+  return client
+}
+
+function setupServerManager(client = setupClient()) {
+  /* eslint-disable-next-line dot-notation */
+  const serverManager = client["_serverManager"]
+  return serverManager
+}
+
 describe("AutoLanguageClient", () => {
+  describe("determineProjectPath", () => {
+    it("returns null when a single file is open", async () => {
+      const client = setupClient()
+      const textEditor = (await atom.workspace.open(__filename)) as TextEditor
+      /* eslint-disable-next-line dot-notation */
+      const projectPath = client["determineProjectPath"](textEditor)
+      expect(projectPath).toBeNull()
+    })
+    it("returns the project path when a file of that project is open", async () => {
+      // macos has issues with handling too much resources
+      if (process.platform === "darwin") {
+        return
+      }
+      const client = setupClient()
+      const serverManager = setupServerManager(client)
+
+      const projectPath = __dirname
+
+      // gives the open workspace folder
+      atom.project.addPath(projectPath)
+      await serverManager.startServer(projectPath)
+
+      const textEditor = (await atom.workspace.open(__filename)) as TextEditor
+      /* eslint-disable-next-line dot-notation */
+      expect(client["determineProjectPath"](textEditor)).toBe(normalizePath(projectPath))
+    })
+    it("returns the project path for an external file if it is in additional paths", async () => {
+      // macos has issues with handling too much resources
+      if (process.platform === "darwin") {
+        return
+      }
```

Comment:
We should mark them as pending so they are easier to track.

```suggestion
      if (process.platform === "darwin") {
        pending("skipped on macos")
        return
      }
```

Reply:
Fixed in 
https://github.com/atom-community/atom-languageclient/pull/160/commits/0f29b94dcf5b0ed49b134b48e49980f7916d8581

Reply:
It seems `pending` cannot be added to describe. So I had to add it only to the `it`s of this PR.